ext/hash/tests/hash_serialize_002.phpt

```diff
@@ -0,0 +1,316 @@
+--TEST--
+Hash: serialize()/unserialize() with HASH_HMAC
+--FILE--
+<?php
+
+$algos = hash_algos();
+$non_crypto = ["adler32", "crc32", "crc32b", "crc32c", "fnv132", "fnv1a32", "fnv164", "fnv1a64", "joaat"];
+$key = "This is the key that I have";
+
+foreach ($algos as $algo) {
+    if (in_array($algo, $non_crypto)) {
+        continue;
+    }
+
+    var_dump($algo);
+    $ctx0 = hash_init($algo, HASH_HMAC, $key);
+    $serial = serialize($ctx0);
```

Comment:
As above, I'd like to include the serialized state in the expect so that we can confirm endianness is accounted for correctly.